src/Microsoft.ML.Core/Utilities/PathUtils.cs

```diff
@@ -139,5 +141,65 @@ public static string CreateFolderIfNotExists(string folder)
 
             return null;
         }
+
+        /// <summary>
+        /// Make a full path realtive to a base path.
+        /// </summary>
+        /// <param name="basepath">The base path, assumed to be a directory.</param>
+        /// <param name="path">The full path.</param>
+        /// <returns>The relative path.</returns>
+        /// <exception cref="ArgumentException">If the paths are not relative.</exception>
+        public static string MakePathRelative(string basepath, string path)
+        {
+            Contracts.AssertNonEmpty(basepath);
+            Contracts.AssertNonEmpty(path);
+
+            Uri baseUri = new Uri(basepath);
+            Uri uri = new Uri(path);
+
+            if (baseUri.Scheme != uri.Scheme)
+            {
+                throw new ArgumentException("Paths cannot be made relative as they are of different schemes.");
+            }
+
+            string relativePath;
+            try
+            {
+                if (!baseUri.AbsoluteUri.EndsWith("/"))
+                {
+                    baseUri = new Uri(baseUri.AbsoluteUri + "/");
+                }
+
+                relativePath = baseUri.MakeRelativeUri(uri).ToString();
+            }
+            catch (ArgumentNullException e)
+            {
+                throw new ArgumentException("Paths could not be made relative.", e);
+            }
+            catch (InvalidOperationException e)
+            {
+                throw new ArgumentException("Paths could not be made relative.", e);
+            }
+
+            if (uri.Scheme.Equals("file", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
>StringComparison.InvariantCultureIgnoreCase [](start = 42, length = 43)

It is generally best practice to use `StringComparison.OrdinalIgnoreCase` for this type of situation. See [here](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings), in particular, the bulleted list under "recommendations for string usage" has some words on when to use ordinal vs. invariant. (The situations where invariant is appropriate are *extremely* limited.)

Reply:
Thanks for the tip.